jdk-1.7-parent/wicketstuff-lazymodel/src/main/java/org/wicketstuff/lazymodel/LazyModel.java

```diff
@@ -556,6 +556,33 @@ private void fillArguments(Object[] args) {
 	}
 
 	/**
+	 * Shortcut for {@link #from(Class)} and {@link #bind(Object)}. Use this for quick model building on top of a
+	 * type-erased model.
+	 *
+	 * @param target The model to bind to.
+	 * @param type The type parameter for that model.
+	 * @param <T>
+	 * @return a result proxy for further evaluation.
+	 * @see #fromObject(IModel)
+	 */
+	public static <T> T from(IModel<T> target, Class<T> type) {
+		return (T) new BoundEvaluation<T>(type, target).proxy();
+	}
+
+	/**
+	 * Start a lazy evaluation, using the reflected runtime type of the given model's object. Use this for terse model
+	 * building on top of a type erased model.
+	 * @param target The model to bind to.
+	 * @param <T>
+	 * @return a result proxy for further evaluation.
+	 * @see #from(IModel, Class)
+	 */
+	public static <T> T fromObject(IModel<T> target)
+	{
+		return (T) new BoundEvaluation<T>(target.getObject().getClass(), target).proxy();
```

Comment:
I do understand the concern. The model object shouldn't be accessed *unless you know what you're doing and why it's bad*. 

The problem is that I'm dealing with a legacy codebase that has stuff like CompoundPropertyModels and plain Models all over the place. I'd like to gradually introduce LazyModel without having to change everything.

Hm. Maybe provide something like eagerlyFrom in a separate class so it would have to be a separate import? Something like "LegacySupport.eagerlyFrom" or so?